pkg/manifest/tekton_builder.go

```diff
@@ -0,0 +1,154 @@
+package manifest
+
+import (
+	"fmt"
+	"net/url"
+	"path/filepath"
+	"strings"
+
+	"github.com/openshift/odo/pkg/manifest/config"
+	"github.com/openshift/odo/pkg/manifest/eventlisteners"
+	res "github.com/openshift/odo/pkg/manifest/resources"
+	"sigs.k8s.io/kustomize/pkg/gvk"
+	"sigs.k8s.io/kustomize/pkg/types"
+)
+
+const (
+	elPatchFile     = "eventlistener_patch.yaml"
+	rolebindingFile = "edit-rolebinding.yaml"
+)
+
+type patchStringValue struct {
+	Op    string      `json:"op"`
+	Path  string      `json:"path"`
+	Value interface{} `json:"value"`
+}
+
+type tektonBuilder struct {
+	files res.Resources
+}
+
+func buildEventlistener(m *config.Manifest) (res.Resources, error) {
+	files := make(res.Resources)
+	tb := &tektonBuilder{files: files}
+	err := m.Walk(tb)
+	return tb.files, err
+}
+
+func (tk *tektonBuilder) Service(env *config.Environment, app *config.Application, svc *config.Service) error {
+
+	svcPath := config.PathForService(env, svc)
+	svcFiles, err := getServiceFiles(svcPath, env, svc)
+	if err != nil {
+		return err
+	}
+	tk.files = res.Merge(svcFiles, tk.files)
+	return nil
+}
+
+func getPipelines(env *config.Environment, svc *config.Service) *config.Pipelines {
+	if svc.Pipelines != nil {
+		return svc.Pipelines
+	}
+	if env.Pipelines != nil {
+		return env.Pipelines
+	}
+	return defaultPipeline()
+}
+
+func defaultPipeline() *config.Pipelines {
+	return &config.Pipelines{
+		Integration: &config.TemplateBinding{
+			Template: "app-ci-template",
+			Binding:  "app-ci-binding",
+		},
+		Deployment: &config.TemplateBinding{
+			Template: "app-cd-template",
```

Comment:
I think we should remove the `Deployment` field from the `Pipelines` struct for now.

It's confusing because we're prioritising ArgoCD (and so, the Deployment pipeline isn't used).

Reply:
I've removed the `Deployment` field from struct